gprofiler/profilers/java.py

```diff
@@ -473,21 +473,24 @@ def _get_stop_cmd(self, with_output: bool) -> List[str]:
             + self._get_extra_ap_args()
         ]
 
+    def _read_ap_log(self) -> str:
+        if not os.path.exists(self._log_path_host):
+            return "(log file doesn't exist)"
+
+        log = Path(self._log_path_host)
+        ap_log = log.read_text()
+        # clean immediately so we don't mix log messages from multiple invocations.
+        # this is also what AP's profiler.sh does.
+        log.unlink()
```

Comment:
small race here for pid recycling, I think, meh

Reply:
Too small